### ISSUE TITLE
Give information on call behaviour

### ISSUE THREAD

**Issue by Jameskmonger** [alsatian-test/alsatian#91]
Currently when an `Expect(...).toHaveBeenCalledWith` fails you get this output:

```
not ok 607 shouldEmitSkipCorrectly
 ---
   message: "Expected function to be called with ["ok 1 Test Function # skip\n"]."
   severity: fail
   data:
     got: undefined
     expect: "function to be called"
 ...
```

Actually the function had been called with `["ok 1 Test Function\n"]`, however the output gave no indication of that.

**Comment by jamesrichford** [comment]
Ah, cool so probably need to clarify the message a bit better between never called and never called with the expected arguments?

**Comment by Jameskmonger** [comment]
Yeah, in the above example I'd have expected this instead:

```
not ok 607 shouldEmitSkipCorrectly
 ---
   message: "Expected function to be called with ["ok 1 Test Function # skip\n"]."
   severity: fail
   data:
     got: "function was called 1 time with ["ok 1 Test Function\n"]"
     expect: "function to be called"
 ...
```

**Comment by jamesrichford** [comment]
@jameskmonger should expect be ```function to be called with ["ok 1 Test Function #skip\n"]```? And then what about if there are multiple calls? Maybe limit output to the most recent 3 arguments e.g. ```function was called 42 times including with ["a"], ["b"] and ["c"]```

**Comment by Jameskmonger** [comment]
Just set up a test fixture for this in Jasmine:

```
var someObject = {
	someFunction: function(arg1, arg2) { }
};

describe("Test spy output", () => {

	it("called with wrong arguments", () => {
		spyOn(someObject, "someFunction");
		
		someObject.someFunction(3, 4);
		someObject.someFunction(5, 6);
		someObject.someFunction(7, 8);
		
		expect(someObject.someFunction).toHaveBeenCalledWith(1, 2);
	});
	
	it("never called", () => {
		spyOn(someObject, "someFunction");
		
		expect(someObject.someFunction).toHaveBeenCalledWith(1, 2);
	});
	
});
```

This is the output:

```
Failures:
1) Test spy output called with wrong arguments
  Message:
    Expected spy someFunction to have been called with [ 1, 2 ] but actual calls were [ 3, 4 ], [ 5, 6 ], [ 7, 8 ].
  Stack:
    Error: Expected spy someFunction to have been called with [ 1, 2 ] but actual calls were [ 3, 4 ], [ 5, 6 ], [ 7, 8 ].
        at Object.<anonymous> (C:\Users\james.monger\Desktop\js\jasmine-spy-output\spec\test.spec.js:14:35)

2) Test spy output never called
  Message:
    Expected spy someFunction to have been called with [ 1, 2 ] but it was never called.
  Stack:
    Error: Expected spy someFunction to have been called with [ 1, 2 ] but it was never called.
        at Object.<anonymous> (C:\Users\james.monger\Desktop\js\jasmine-spy-output\spec\test.spec.js:20:35)
```

**Comment by jamesrichford** [comment]
To be solved in #155

**Comment by jamesrichford** [comment]
closed by #167
